src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -31,7 +32,7 @@ internal partial class DesignerAttributeIncrementalAnalyzer : ForegroundThreadAf
         private readonly IAsynchronousOperationListener _listener;
 
         // cache whether a project is cps project or not
-        private readonly ConcurrentDictionary<ProjectId, bool> _cpsProjects;
+        private readonly ConcurrentDictionary<ProjectId, IProjectItemDesignerTypeUpdateService> _cpsProjects;
```

Comment:
You must release this object, when the project is closed.  Otherwise, it may leak the entire project (through MEF).

Reply:
we already let it go when product is removed
https://github.com/dotnet/roslyn/blob/28ce564d7d4ce9e6adedd1848705917cfc7093dd/src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs#L117